infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -106,7 +108,11 @@
     public LogUnitServer(ServerContext serverContext) {
         this.opts = serverContext.getServerConfig();
         this.serverContext = serverContext;
-        maxCacheSize = Utils.parseLong(opts.get("--max-cache"));
+        long maxCacheSizeFromUser = Utils.parseLong(opts.get("--max-cache"));
+
+        /* Ensure that maxCacheSize is at most CACHE_HEAP_RATIO * maxHeapSize */
```

Comment:
It prevents the user to specify a cache that is too big for the heap. If we let the user do this, it will end up in OutOfMemory exception. The cache heap ratio is the biggest ratio that deem safe to use.